rust/rsc/src/rsc/read_job.rs

```diff
@@ -44,12 +75,29 @@ pub async fn read_job(
                     .all(txn)
                     .await?
                     .into_iter()
-                    .map(|m| File {
-                        path: m.path,
-                        mode: m.mode,
-                        blob_id: m.blob_id,
-                    })
-                    .collect();
+                    .map(|m| {
+                        let blob_copy = blob_stores.clone();
+                        async move {
+                            let blob = resolve_blob(m.blob_id, txn, &blob_copy).await?;
+
+                            Ok(ResolvedBlobFile {
+                                path: m.path,
+                                mode: m.mode,
+                                blob,
+                            })
+                        }
+                    });
+
+                let output_files: Result<Vec<ResolvedBlobFile>, String> =
+                    futures::future::join_all(output_files)
+                        .await
+                        .into_iter()
+                        .collect();
+
+                let output_files = match output_files {
+                    Err(_) => return Ok((None, ReadJobResponse::NoMatch)),
```

Comment:
Do we really want to have the same response for a failure to find blobs of a known job as we do for not finding a job?
I think from the client perspective it makes sense that in neither of these cases can we use the RSC for this job, but in the bad blob case there is no trail on the client that anything went wrong even though any number of things could have gone wrong AFAICT.

Reply:
I agree this response isn't super great. This was me just not wanting to massage rust to be happy here. I'll see what I can do. Mainly I still want to return NoMatch, but maybe I can set a status code or something so that we get an idea that something went wrong

Reply:
In scenarios where we are debugging an RSC read failure, will there be any log/record of the `Err` that is hidden on this line?

Reply:
I can add a log super easily